File: python-pr/52/python/code-quality-performance/string_counter_20.py
# {fact rule=code-quality-performance@v1.0 defects=1}
def count_characters(text):
    counts = {}
    for char in text:
        if char in counts:
            counts[char] += 1
        else:
            counts[char] = 1
    return counts
# {/fact}

File: js-pr/56/javascript/code-quality-performance/string_counter_20.js
// {fact rule=code-quality-performance@v1.0 defects=1}
function countCharacters(text) {
    let counts = {};
    for (let char of text) {
        if (counts[char]) {
            counts[char]++;
        } else {
            counts[char] = 1;
        }
    }
    return counts;
}
// {/fact}